src/data_handlers/user_data_handler.js

```diff
@@ -68,20 +34,12 @@ class UserDataHandler {
     return isMatching
```

Comment:
Let's changes to UserDataHandler class after you have 100% coverage by unit tests, and see if any of the test fail

```
    return Object.keys(searchParamsObject).some((searchParam) => user[searchParam] === searchParamsObject[searchParam]);
```